press/views/legacy/redirects.py

```diff
@@ -0,0 +1,61 @@
+from pyramid import httpexceptions
+from pyramid.view import view_config
+from sqlalchemy.sql import desc
+
+from press.utils import convert_to_legacy_domain
+
+
+def _lookup_latest_legacy_version(request, legacy_id):
+    """Given the legacy id for content, lookup the latest version."""
+    t = request.db_tables
+    with request.get_db_engine('common').begin() as db_conn:
+        results = db_conn.execute(
+            t.latest_modules.select()
+            .where(t.latest_modules.c.moduleid == legacy_id)
+        )
+        return getattr(results.fetchone(), 'version', None)
+
+
+def _lookup_head_legacy_version(request, legacy_id):
+    """Given the legacy id for content, lookup the very latest version."""
+    t = request.db_tables
+    with request.get_db_engine('common').begin() as db_conn:
+        results = db_conn.execute(
+            t.modules.select()
+            .where(t.modules.c.moduleid == legacy_id)
+            .order_by(desc('module_ident'))
```

Comment:
I don't like to depend on the monotonically increasing nature of module_ident - it's not actually guaranteed in a multi-client environment, since it's filled from a sequence which doles out sets of numbers to different simultaneously connected clients - so if C1  and C2 are publishing, then may get C1 = [1,2,3,4,5], C2 = [6,7,8,9,10], but if C1 publishes 1, then waits, and C2 publishes, then C1 again, you could have in actual time order: 1, 6, 2 …

It'll probably never bite us, so we an go with this for now.